db/dcrpg/pgblockchain.go

```diff
@@ -66,6 +67,57 @@ func (d *DevFundBalance) Balance() *explorer.AddressBalance {
 	return d.AddressBalance
 }
 
+// ticketPoolDataCache stores the most recent ticketpool graphs information
+// fetched to minimize the possibility of making multiple queries to the db
+// fetching the same information.
+type ticketPoolDataCache struct {
+	sync.RWMutex
+	Height map[dbtypes.ChartGrouping]uint64
+	// BarGraphsCache persists data for the Ticket purchase distribution chart
+	// and Ticket Price Distribution chart
+	BarGraphsCache map[dbtypes.ChartGrouping][]*dbtypes.PoolTicketsData
+	// DonutGraphCache persist data for the Number of tickets outputs pie chart.
+	DonutGraphCache map[dbtypes.ChartGrouping]*dbtypes.PoolTicketsData
+}
+
+// ticketPoolGraphsCache persists the latest ticketpool data queried from the db.
+var ticketPoolGraphsCache = &ticketPoolDataCache{
+	Height:          make(map[dbtypes.ChartGrouping]uint64),
+	BarGraphsCache:  make(map[dbtypes.ChartGrouping][]*dbtypes.PoolTicketsData),
+	DonutGraphCache: make(map[dbtypes.ChartGrouping]*dbtypes.PoolTicketsData),
+}
+
+// TicketPoolData is a thread-safe way to access the ticketpool graphs data
+// stored in the cache.
+func TicketPoolData(interval dbtypes.ChartGrouping, height uint64) (barGraphs []*dbtypes.PoolTicketsData,
+	donutChart *dbtypes.PoolTicketsData, actualHeight uint64, intervalFound, isStale bool) {
+	ticketPoolGraphsCache.RLock()
+	defer ticketPoolGraphsCache.RUnlock()
+
+	var found bool
+	barGraphs, intervalFound = ticketPoolGraphsCache.BarGraphsCache[interval]
+	donutChart, found = ticketPoolGraphsCache.DonutGraphCache[interval]
+	intervalFound = intervalFound && found
+
+	actualHeight = ticketPoolGraphsCache.Height[interval]
+	isStale = ticketPoolGraphsCache.Height[interval] != height
+
+	return
+}
+
+// UpdateTicketPoolData updates the ticket pool cache with the latest data fetched.
+// This is a thread-safe way to update ticket pool cache data. TryLock helps avoid
+// stacking calls to update the cache.
+func UpdateTicketPoolData(interval dbtypes.ChartGrouping, barGraphs []*dbtypes.PoolTicketsData,
+	donutcharts *dbtypes.PoolTicketsData, height uint64) {
+	ticketPoolGraphsCache.Lock()
+	defer ticketPoolGraphsCache.Unlock()
+
+	ticketPoolGraphsCache.Height[interval] = height
```

Comment:
I don't know if you've got a mechanism to ensure that the `height` map never grows to uncontrollable sizes because it will always be a unique entry that you make here every time.

Reply:
Since the key is `interval` and there are only a handful of intervals, it is not going to grow unbounded.